### PR TITLE
chore(deps): Implemented best practices for bash scripts.

### PR DESCRIPTION
**Pull request by robertvolkmann**
### Changes

* Implemented best practices for bash scripts.
* Removed Mandriva stuff, because it is EOL.
* Removed empty mod_deps.
* Removed unnecessary checks for YesOpt.
* Replaced unrar with unrar-free for debian jessy. (unrar is in the non-free repo)

If any command failed, then the fo-installdeps exited with a non-zero exit code.

edit: Fixed docker-compose.Dockerfile.

### CODE REVIEW

**Comment by mcjaeger** [comment]
very nice. I guess we will have some (minor) conflicts with #1116

**Comment by robertvolkmann** [comment]
@mcjaeger no problem. I will adjust this pull request after #1116 was merged.

**Comment by maxhbr** [comment]
depends on #1116

**Comment by maxhbr** [comment]
> If any command failed, then the fo-installdeps exited with a non-zero exit code.

this is a very important and good change!

File: utils/utils.sh [hunk 1]
```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env bash
```
Comment (maxhbr) [inline utils/utils.sh]: Missing license header

Comment (robertvolkmann) [reply utils/utils.sh]: Done.

**Comment by GMishx** [comment]
The option `set -o errexit` is causing issues in Ubuntu 16.04.
```
$ sudo ./utils/fo-installdeps 
*** Installing both runtime and buildtime dependencies ***
install core dependencies
*** Installing Ubuntu buildtime dependencies ***
DB: Installing build essential.....
E: Invalid operation 
```
Whereas, passing `-y -e` flags skips the errors.
Bash version: 4.3.48

**Comment by robertvolkmann** [comment]
@GMishx I have created the pull request #1189 to fix it together with other issues.

**Pull request merged**